### PR TITLE
Remotor

### PR DESCRIPTION
**Pull request by Tasssadar**
You can now get a motor instance and use it. The encoders are now members of that Motor class.

### CODE REVIEW

File: src/RBControl_manager.hpp [hunk 1]
```diff
@@ -187,7 +169,7 @@ class MotorChangeBuilder {
      * \param id of the motor (e.g. rb:MotorId::M1)
      * \param percent of the maximal power of the motor <0 - 100>
      **/
-    MotorChangeBuilder& pwmMaxPercent(MotorId id, uint8_t percent); 
+    MotorChangeBuilder& pwmMaxPercent(MotorId id, int8_t percent);
```
Comment (JarekParal) [inline src/RBControl_manager.hpp:172]: Why you change the type of parameter `percent` from `uint8_t` to `int8_t`. Isn't the `uint8_t` safer (`<0 - 100>`)?

File: src/RBControl_motor.hpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+     * \brief Limit the maximum PWM value. If you call pwmMaxPercent(70) and then
+     * power(100), the motors will spin at 70% of maximum speed.
+     * \param pct of the max value <0 - 100>
+     */
+    void pwmMaxPercent(int8_t percent);
+
+    /**
+     * \brief Drive motor to set position (according absolute value). See {@link Encoder::driveToValue}. 
+     */
+    void driveToValue(int32_t positionAbsolute, uint8_t power, EncoderDoneCallback callback = nullptr, void *cookie = nullptr);
```
Comment (JarekParal) [inline src/RBControl_motor.hpp:36]: When we have this method in the `Motor` class do we need it also in the `Encoder` class?

Or should it be public in the `Encoder` class?

File: src/RBControl_manager.cpp [hunk 3]
```diff
@@ -189,23 +196,6 @@ bool Manager::motorsFailSafe(void *cookie) {
     return true;
 }
 
-void Manager::initEncoder(MotorId id) {
```
Comment (JarekParal) [inline src/RBControl_manager.cpp:192]: Don't we need now this method? The encoder is automatically initialized in `Motor` constructor or the user must use some other method?

Comment (JarekParal) [reply src/RBControl_manager.cpp:192]: I have already found the actual [solution](https://github.com/RoboticsBrno/RB3201-RBControl-library/pull/9/files#diff-d219de94c23b39cd5045d2fbc0db8b50R71).

In the new version you firstly check, when you want to get the encoder instance, you firstly check, if is the encoder initialize and when not you initialize it. Am I correct?

**Review by JarekParal — COMMENTED** [review]
Could you please look at my comments? @Tasssadar

File: src/RBControl_manager.hpp (hunk 1 above)
Comment (Tasssadar) [reply src/RBControl_manager.hpp:172]: It is, but I need the two methods (power and pwmMaxPercent) to have the same signature in order to have pointer to them in Manager.

File: src/RBControl_motor.hpp (hunk 2 above)
Comment (Tasssadar) [reply src/RBControl_motor.hpp:36]: Currently, it is both, this is just a shortcut.

File: src/RBControl_manager.cpp (hunk 3 above)
Comment (Tasssadar) [reply src/RBControl_manager.cpp:192]: That is correct.

**Pull request merged**